### PR TITLE
Bug 1354522, describe parameters mentioned in master-config.yaml

### PR DESCRIPTION
**Pull request by ahardin-rh**
https://bugzilla.redhat.com/show_bug.cgi?id=1354522

### CODE REVIEW

File: install_config/master_node_configuration.adoc [hunk 1]
```diff
@@ -33,259 +33,406 @@ remember all of the options to specify. To make this easier, the configuration
...
+
+|`*disabledFeatures*`
+|A list of features that should not be started. You will likely want to set this
+as *null*. It is very unlikely that anyone will want to manually disable
+features and that is not encouraged.
+
+|===
+
+
+=== etc Configuration
```
Comment (adellape) [inline install_config/master_node_configuration.adoc]: s/etc/etcd/

and below

**Comment by ahardin-rh** [comment]
Pretty Build: http://file.rdu.redhat.com/~ahardin/09022016/master-config-files/install_config/master_node_configuration.html#master-config-admission-control-config

@sferich888 PTAL Thanks!

**Comment by ahardin-rh** [comment]
@bfallonf @tpoitras peer review please. Thanks! :bow:

File: install_config/master_node_configuration.adoc [hunk 2]
```diff
@@ -33,259 +33,698 @@ remember all of the options to specify. To make this easier, the configuration
...
+|Key value pairs that will be passed directly to the Kube API server that match
+the API servers' command line arguments. These are not migrated, but if you
+reference a value that does not exist the server will not start. These values
+may override other settings in `*KubernetesMasterConfig*`, which may cause
+invalid configurations.
 
-To create a xref:node-configuration-files[node configuration file] and other
-related files in the specified directory:
+|`*ControllerArguments*`
+|Key value pairs that will be passed directly to the Kube controller manager
```
Comment (bfallonf) [inline install_config/master_node_configuration.adoc]: @ahardin-rh I just did a search, and sometimes in the docs we use "key-value". I'm not sure which is more accurate, but I'd lean towards that one. We're pretty loose with this though, so I'd say both are acceptable. Developing a standard might be a good idea for the future (brice thinks to himself...).

File: install_config/master_node_configuration.adoc [hunk 3]
```diff
@@ -33,259 +33,698 @@ remember all of the options to specify. To make this easier, the configuration
...
+migrated, but if you reference a value that does not exist the server will not
+start. These values may override other settings in `*KubernetesMasterConfig*`,
+which may cause invalid configurations.
 
-[options="nowrap"]
-----
-$ oadm create-node-config --node-dir=/openshift.local.config/node-<node_hostname> --node=<node_hostname> --hostnames=<hostname>,<ip_address>
-----
+|`*DefaultAdmissionConfig*`
+|Used to enable or disable various admission plug-ins.When this type is present
```
Comment (bfallonf) [inline install_config/master_node_configuration.adoc]: Space before 'when'?

File: install_config/master_node_configuration.adoc [hunk 4]
```diff
@@ -33,259 +33,698 @@ remember all of the options to specify. To make this easier, the configuration
...
-above command also assumes that certificate files are located in an
-*_openshift.local.config/master/_* directory. If they are not, you can include
-options to specify their location. Run the command with the `-h` option to see
-details.
+|`*PluginConfig*`
+|Allows specifying a configuration file per admission control plug-in.
 
-[[launching-servers-using-configuration-files]]
+|`*PluginOrderOverride*`
+|A list of admission control plugin names that will be installed on the master.
```
Comment (bfallonf) [inline install_config/master_node_configuration.adoc]: s/plugin/plug-in

File: install_config/master_node_configuration.adoc [hunk 5]
```diff
@@ -33,259 +33,698 @@ remember all of the options to specify. To make this easier, the configuration
...
-== Launching Servers Using Configuration Files
-Once you have modified the master and/or node configuration files to your
-specifications, you can use them when launching servers by specifying them as an
-argument. Keep in mind that if you specify a configuration file, none of the
-other command line options you pass are respected.
+|`*SchedulerArguments*`
+|Key value pairs that will be passed directly to the Kube scheduler that match
+the scheduler's command line arguments. These are not migrated, but if you
+reference a value that does not exist the server will not start. These values
+may override other settings in `*KubernetesMasterConfig*` which may cause
```
Comment (bfallonf) [inline install_config/master_node_configuration.adoc]: Comma before 'which'.

File: install_config/master_node_configuration.adoc [hunk 6]
```diff
@@ -33,259 +33,698 @@ remember all of the options to specify. To make this easier, the configuration
...
+=== Controller Configuration
+
+.Controller Configuration Parameters
+[cols="3a,6a",options="header"]
+|===
+
+| Parameter Name | Description
+
+|`*Controllers*`
+|list of the controllers that should be started. If set to *none*, no
```
Comment (bfallonf) [inline install_config/master_node_configuration.adoc]: Caps for "list".

File: install_config/master_node_configuration.adoc [hunk 7]
```diff
@@ -33,259 +33,698 @@ remember all of the options to specify. To make this easier, the configuration
...
+
+|`*ControllerLeaseTTL*`
+|Enables controller election, instructing the master to attempt to acquire a
+lease before controllers start and renewing it within a number of seconds
+defined by this value. Setting this value non-negative forces
+`*pauseControllers=true*`. This value defaults off (0, or omitted) and controller
+election can be disabled with -1.
+
+|`*PauseControllers*`
+|Instructs the master to not automatically start controllers, but instead  to
```
Comment (bfallonf) [inline install_config/master_node_configuration.adoc]: double space after 'instead'.

File: install_config/master_node_configuration.adoc [hunk 8]
```diff
@@ -33,259 +33,698 @@ remember all of the options to specify. To make this easier, the configuration
...
+|`*KubernetesMasterConfig*`
+|Holds the necessary configuration options for the Kubernetes master.
+
+|`*MasterCount*`
+|The number of expected masters that should be running. This value defaults to 1
+and may be set to a positive integer, or if set to -1, indicates this is part of
+a cluster.
+
+|`*MasterIP*`
+|The  public IP address of Kubernetes resources. If empty, the first result from
```
Comment (bfallonf) [inline install_config/master_node_configuration.adoc]: Double space before 'public'.

File: install_config/master_node_configuration.adoc [hunk 9]
```diff
@@ -33,259 +33,698 @@ remember all of the options to specify. To make this easier, the configuration
...
+
+|`*CertInfo*`
+|TLS cert information for serving secure traffic.
+
+|`*ClientCA*`
+|The certificate bundle for all the signers that you recognize for incoming
+client certificates.
+
+|`*dnsConfig*`
+|holds the necessary configuration options for DNS.
```
Comment (bfallonf) [inline install_config/master_node_configuration.adoc]: Caps for 'holds'.

File: install_config/master_node_configuration.adoc [hunk 10]
```diff
@@ -33,259 +33,698 @@ remember all of the options to specify. To make this easier, the configuration
...
+=== Volume Configuration
+
+.Volume Configuration Parameters
+[cols="3a,6a",options="header"]
+|===
+
+| Parameter Name | Description
+
+|`*DynamicProvisioningEnabled*`
+|A boolean that toggles dynamic provisioning off when false, defaults to true.
```
Comment (bfallonf) [inline install_config/master_node_configuration.adoc]: I'm not sure the above is saying as intended. Maybe just:

"A boolean to enable or disable dynamic provisioning. Default is true."

**Comment by bfallonf** [comment]
@ahardin-rh Some typo-level comments. Overall, looks good!

**Comment by ahardin-rh** [comment]
@bfallonf Thanks for the review! :mag:

File: install_config/master_node_configuration.adoc [hunk 11]
```diff
@@ -33,259 +33,698 @@ remember all of the options to specify. To make this easier, the configuration
...
+== Master Configuration Files
+This section reviews parameters mentioned in the *_master-config.yaml_* file.
 
-== Creating New Configuration Files
-For masters, the `openshift start` command accepts options that indicate that it
-should simply write the configuration files that it would have used, then
-terminate. For nodes, a configuration file can be written using the `oadm
-create-node-config` command. Creating new configuration files is useful to get a
-starting point for defining your configuration.
+You can xref:creating-new-configuration-files[create a new node configuration
```
Comment (mdshuai) [inline install_config/master_node_configuration.adoc]: "create a new master configuration file", not node.

File: install_config/master_node_configuration.adoc [hunk 12]
```diff
@@ -33,259 +33,698 @@ remember all of the options to specify. To make this easier, the configuration
...
+election can be disabled with -1.
+
+|`*PauseControllers*`
+|Instructs the master to not automatically start controllers, but instead to
+wait until a notification to the server is received before launching them.
+
+|===
+
+[[master-config-docker-config]]
+=== Docker Configuration
```
Comment (mdshuai) [inline install_config/master_node_configuration.adoc]: Docker Configuration is in node-config.yaml.

File: install_config/master_node_configuration.adoc [hunk 13]
```diff
@@ -33,259 +33,698 @@ remember all of the options to specify. To make this easier, the configuration
...
+
+|`*NodeConfig*`
+|The fully specified configuration starting an {product-title} node.
+
+|`*NodeIP*`
+|Node may have multiple IPs, so this specifies the IP to use for pod traffic
+routing. If not specified, network parse/lookup on the *nodeName* is performed
+and the first non-loopback address is used.
+
+|`*NodeName*`
```
Comment (mdshuai) [inline install_config/master_node_configuration.adoc]: same above

File: install_config/master_node_configuration.adoc [hunk 14]
```diff
@@ -33,259 +33,698 @@ remember all of the options to specify. To make this easier, the configuration
...
+.Pod and Node Configuration Parameters
+[cols="3a,6a",options="header"]
+|===
+
+| Parameter Name | Description
+
+|`*NodeConfig*`
+|The fully specified configuration starting an {product-title} node.
+
+|`*NodeIP*`
```
Comment (mdshuai) [inline install_config/master_node_configuration.adoc]: This is in in node-config.yaml.

**Comment by mdshuai** [comment]
Is there any plan to describe parameters mentioned in node-config.yaml?thanks

**Comment by ahardin-rh** [comment]
@mdshuai Thanks for your comments. This is updated. Yes, I think node-config.yaml parameters will need to be expanded in a separate PR, although I pulled some relevant parameters into that section with these edits.

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../install_config/master_node_configuration.adoc#install-config-master-node-configuration[Master and Node Configuration]
|Described parameters mentioned in xref:../install_config/master_node_configuration.adoc#master-configuration-files[*_master-config.yaml_*].
%

**Pull request merged**